master/cmd/determined-master/root.go

```diff
@@ -137,6 +143,35 @@ func getConfig(configMap map[string]interface{}) (*internal.Config, error) {
 	return config, nil
 }
 
+func applyBindMount(configMap map[string]interface{}) (map[string]interface{}, error) {
+	bindMountField, fieldExists := configMap["auto_bind_mount"]
+	if !fieldExists {
+		return configMap, nil
+	}
+	vBindMount, ok := bindMountField.(string)
+	if !ok {
+		return nil, errors.New("wrong type for bind mount field")
+	}
+
+	if vBindMount != "no" {
+		_, tcdExisted := configMap["task_container_defaults"]
```

Comment:
I am not sure I quite like this, an env variable which overwrites a chunk of config may be a bit fragile.
We've talked about the `master.yaml.template` for aws & gcp last week. Maybe we should adopt a similar approach for det deploy local: instead of passing a set of env variables through docker-compose, have a `master.yaml.template` inside `det deploy` cli, which is filled in with values, and then mounted into the local master container as a file.

having said that, it may be a bigger and more radical change, so I won't block this PR over it.

Reply:
Agreed. Don't think this PR should be blocked. Once we figure out what library or template language to use in that, we could move this environment variable to the new `master.yaml.template`.